### ISSUE TITLE
[RyuJIT/arm32][JitStressRegs=8] Assertion failed 'this->gtOper == GT_LIST || this->gtOper == GT_FIELD_LIST'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#14551]
From corefx x86_arm_altjit run, e.g. "@dotnet-bot test Windows_NT x86_arm_altjit Checked corefx_jitstressregs8".

Happens with `COMPlus_JitStressRegs=8`.

Found in System.Diagnostics.FileVersionInfo.Tests:
```
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>RunTests.cmd E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\
Using E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\ as the test runtime folder.
Executing in E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86\
Running tests... Start time: 14:44:34.59

E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJit=*
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitNgen=*
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitName=protononjit.dll
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_NoGuiOnAssert=1
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_ContinueOnAssert=0
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitAssertOnNYI=1
E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_JitStressRegs=8

E:\gh\corefx\bin\tests\System.Diagnostics.FileVersionInfo.Tests\netcoreapp-Windows_NT-Release-x86>call E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\\dotnet.exe xunit.console.netcore.exe System.Diagnostics.FileVersionInfo.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonwindowstests  -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Diagnostics.FileVersionInfo.Tests
Discovered:  System.Diagnostics.FileVersionInfo.Tests
Starting:    System.Diagnostics.FileVersionInfo.Tests

Assert failure(PID 23496 [0x00005bc8], Thread: 10464 [0x28e0]): Assertion failed 'this->gtOper == GT_LIST || this->gtOper == GT_FIELD_LIST' in 'System.Diagnostics.FileVersionInfo:GetVersionInfoForCodePage(int,ref):bool:this' (IL size 612)

    File: e:\gh\coreclr2\src\jit\gtstructs.h Line: 68
    Image: E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\dotnet.exe
```

Full repro:
```
dotnet/coreclr:
E:\gh\coreclr2> build.cmd x86 checked

dotnet/corefx:
E:\gh\corefx> build.cmd -Release -os:Windows_NT -buildArch:x86 -- /p:CoreCLROverridePath=E:\gh\coreclr2\bin\Product\Windows_NT.x86.Checked
E:\gh\corefx> build-tests.cmd -Release -os:Windows_NT -buildArch:x86 -- /p:WithoutCategories=IgnoreForCI /p:PreExecutionTestScript=e:\bugs\SetStressModes.bat

With e:\bugs\SetStressModes.bat:
set COMPlus_AltJit=*
set COMPlus_AltJitNgen=*
set COMPlus_AltJitName=protononjit.dll
set COMPlus_NoGuiOnAssert=1
set COMPlus_ContinueOnAssert=0
set COMPlus_AltJitAssertOnNYI=1
set COMPlus_JitStressRegs=8
```

**Comment by BruceForstall** [comment]
@dotnet/arm32-contrib

**Comment by YongseopKim** [comment]
On linux/arm32,
```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  ./run_fxutc_ryujit_arm.sh 
LOG: FX_RYUJIT_10191407.log
export COMPlus_AltJit="*"
export COMPlus_AltJit="*"
export COMPlus_AltJitNgen="*"
#export COMPlus_AltJitName="protononjit.dll"
export COMPlus_NoGuiOnAssert=1
export COMPlus_ContinueOnAssert=0
export COMPlus_AltJitAssertOnNYI=1
export COMPlus_JitStress=8
/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/run-test.sh --runtime /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/overlay --corefx-tests /home/dragon/dotnet/arm32-dotnet/tmp/coreoverl
ay-ARM-CHE-REL-10181724/tests --os Linux --arch x64 --configurationGroup Release --test-dir-file ./tests_fx2.txt

Running tests in /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/tests/System.Diagnostics.FileVersionInfo.Tests/netcoreapp-Linux-Release-x64
./RunTests.sh /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/overlay

Using as the test runtime folder.
Running tests... Start time: 14:07:22
Commands:
export XUNIT_PERFORMANCE_MIN_ITERATION=1
export XUNIT_PERFORMANCE_MAX_ITERATION=1
chmod +x /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/overlay/dotnet
/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/overlay/dotnet xunit.console.netcore.exe System.Diagnostics.FileVersionInfo.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait categor
y=nonlinuxtests  -notrait category=failing
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/tests/System.Diagnostics.FileVersionInfo.Tests/netcoreapp-Linux-Release-x64 ~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/tests/System.Diagnostics.FileVersionI
nfo.Tests/netcoreapp-Linux-Release-x64
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Diagnostics.FileVersionInfo.Tests
Discovered:  System.Diagnostics.FileVersionInfo.Tests
Starting:    System.Diagnostics.FileVersionInfo.Tests
Finished:    System.Diagnostics.FileVersionInfo.Tests

=== TEST EXECUTION SUMMARY ===
   System.Diagnostics.FileVersionInfo.Tests  Total: 7, Errors: 0, Failed: 0, Skipped: 0, Time: 10.214s
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/tests/System.Diagnostics.FileVersionInfo.Tests/netcoreapp-Linux-Release-x64
Finished running tests. End time=14:07:56. Return value was 0
All tests passed.
```

**Comment by YongseopKim** [comment]
```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  cat tmp/coreoverlay-ARM-CHE-REL-10181724/COMMITS.CLR 
e869941a7364ef128a50ecdfce0a9a0e317d13c2 format spmi sources. (#14545)
a194ac80eb0e97e822b9eb35c24c59c3c24d7693 Merge pull request #14553 from briansull/fix-warn
882b1d64a179f2d04dc66c8845b723f06c30bdea Merge pull request #14555 from wtgodbe/AzureBlobSymPkg
de4559f6dbaa7fd732f6529e93b0aebde0fb0ea9 Use C#-friendly path for indexing symbol packages
d183b9f0cb9235440902eb6f80841830d96f5775 Fix for x86 desktop build Fix warning C4389: '==' : signed/unsigned mismatch
4fa06e6095af6473d2ec6c855ebfb8d77017d91d Merge pull request #14541 from stephentoub/taskbeginwait
2f01c285d1a2d01cf9d8810e557961872c1eb46d GetEnvironmentVariable: Avoid StringBuilder marshaling overhead (#14502)
2cd4ca4cdd8b71563a58835850aa67de5906821c CLRLifoSemaphore cleanup (#14535)
0526339a302da4918699901d5fc403044374651b Avoid TaskWaitBegin/End events in Task.Wait on already completed task
b5616f0013ccba23c6a3e6facbb09545e710a5f1 Merge pull request #14493 from alpencolt/armel-cross-build
```

```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  cat tmp/coreoverlay-ARM-CHE-REL-10181724/COMMITS.FX 
5c8af507d63e94a49ab4b4a81a45510a50146faf Adding ValidateTypeIsBlittable SpanHelper method and rename constant (#24626)
b28112eb892283f20e7e422a52da0a55ca4881d5 Rewrite BackgroundWorker test to be non blocking (#24693)
ae86eca620025087b576d092e67acde7760d4baa Deleted unused string resources. (#24697)
a1e7009bd3c94c3999bfd471a08e5d647c571e52 Add null check when setting Packaging property (#24673)
0ad10b8bf74a4ce2fe452aa23801b3722237560d Update CoreSetup, Standard to preview1-25817-01, preview1-25817-02, respectively (#24702)
5ef5f17b5606377415d5c171d6613c49c59a1b55 Set.TryGetValue not-found behavior/docs inconsistent (#24699)
79089260b0bf00e99ba9dd7e736c7b2d776fed0a Deleted dead code from System.IO.FileSystem. (#24701)
3b867880d9992f305757db85a9b32044f9773863 Fix for GraphicsPathIterator.CopyData throws EntrypointNotFoundException when index is out of range. (#24656)
e3977683cf78b7b1896a11e9de7df6a62408a1a6 Remove methods marked as internal from reference assembly. (#24700)
db596d27932201ec4936a7c9baa6aa6ad3f9e796 Update XunitNetcoreExtensionsVersion to 1.0.1-prerelease-02116-01 (#24687)
```

**Comment by BruceForstall** [comment]
@YongseopKim You used:
```
export COMPlus_JitStress=8
```

you need to use:
```
export COMPlus_JitStressRegs=8
```

**Comment by YongseopKim** [comment]
@BruceForstall I used `export COMPlus_JitStressRegs=8` and the result is that the tc passed

```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  ./run_fxutc_ryujit_arm.sh 
LOG: FX_RYUJIT_10241509.log
export COMPlus_AltJit="*"
export COMPlus_AltJitNgen="*"
#export COMPlus_AltJitName="protononjit.dll"
export COMPlus_NoGuiOnAssert=1
export COMPlus_ContinueOnAssert=0
export COMPlus_AltJitAssertOnNYI=1
export COMPlus_JitStressRegs=8

/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/run-test.sh --runtime /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay --corefx-tests /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests --os Linux --arch x64 --configurationGroup Release --test-dir-file ./tests_fx2.txt

Running tests in /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests/System.Diagnostics.FileVersionInfo.Tests/netcoreapp-Linux-Release-x64
./RunTests.sh /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay

Using as the test runtime folder.
Running tests... Start time: 15:09:44
Commands:
export XUNIT_PERFORMANCE_MIN_ITERATION=1
export XUNIT_PERFORMANCE_MAX_ITERATION=1
chmod +x /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay/dotnet
/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay/dotnet xunit.console.netcore.exe System.Diagnostics.FileVersionInfo.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests  -notrait category=failing
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests/System.Diagnostics.FileVersionInfo.Tests/netcoreapp-Linux-Release-x64 ~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests/System.Diagnostics.FileVersionInfo.Tests/netcoreapp-Linux-Release-x64
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Diagnostics.FileVersionInfo.Tests
Discovered:  System.Diagnostics.FileVersionInfo.Tests
Starting:    System.Diagnostics.FileVersionInfo.Tests
Finished:    System.Diagnostics.FileVersionInfo.Tests

=== TEST EXECUTION SUMMARY ===
   System.Diagnostics.FileVersionInfo.Tests  Total: 7, Errors: 0, Failed: 0, Skipped: 0, Time: 8.833s
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests/System.Diagnostics.FileVersionInfo.Tests/netcoreapp-Linux-Release-x64
Finished running tests. End time=15:10:16. Return value was 0
All tests passed.
```

```
 dragon@loki  ~/dotnet/arm32-dotnet/tmp  cat coreoverlay-ARM-CHE-REL-10241504/COMMITS.CLR 
2207dadf19fc976c6ccd04e12d46924a7b733457 Report registers as dead in GCInfo before the RhpPInvoke helper. (#14664)
5d66791a3a1932d2afcd3ef58dd0df91323662cb profiler changes for tiered compilation (#14612)
f8d6405c8e0680a1184dc07123aeb4abe62e9e87 Merge pull request #14647 from BruceForstall/AddArm64FrameLayoutDoc
67a66156d0307b49522a13618522d289268bc1f9 Fix for #12609 - add option to disable tiered compilation for profilers  (#14643)
33159233fd161877ffe37187bf06689fe0e8bea2 Fix RID extraction in packages and test build for Alpine (#14656)
458c90a363b22fb29f6581610d5349bf06e17971 Merge pull request #14649 from mikedn/float-absneg
2ba74317c027eb4ff1e854de02fbb9378d00a34e [Arm64] SIMD simple defines (#14628)
c0aa11ff835222facec5401c4074e5b46cb1635f [Arm64] SIMD lsra (#14631)
e9437e8a686d4db2a6667eb7f386279678916467 [Arm64] SIMD ins_Load (#14636)
4d253390b02d51dd5dc2049dbe83c122c3acba9d Merge pull request #14528 from jashook/pri1_rework

 dragon@loki  ~/dotnet/arm32-dotnet/tmp  cat coreoverlay-ARM-CHE-REL-10241504/COMMITS.FX 
7e91c485254e054e0f3c14291f97e6a382659bec Fix condition in System.Diagnostics.PerformanceCounter (#24828)
bcc59b2e74090cfbcc4d6e7c245c172744e9d46b Fix System.Data.Odbc configurations for support package and run UpdateVSConfigurations (#24752)
389d7ee0630e71ff868b4d9ebcdcf8111d5acee3 Merge pull request #24684 from Anipik/newbr
5d164eb843b6fc2278c107184634396642483901 Re-enabling previously disabled tests due to Retain bug (#24785)
1eabfe92ef9fe97f8abfccb044fd6bcfffa5bb5f Add some System.Collections.Concurrent perf tests (#24810)
a0c65af98df8fe746bb49da90b6c2a152d747a18 Remove New APIs in Syndication. (#24814)
d33e40e27e40d313977d324ae89c33f2842ee5b7 Deleted dead code from System.Net.Requests. (#24815)
c92789464b2ac983eba3a6021b13ef07eb0c8d8d Extentionclass added
725495577f7e17f1ae2ad2439f219203b2cef844 Add perf test to measure Memory.Span.Slice vs. Memory.Slice.Span (#24780)
90edf7d04a1d32266c275aa5b239b6022ebab98b Change Memory.Slice(...).Span to Memory.Span.Slice(...) (#24813)
```

It also passed if coreoverlay's clr is built with release mode.
